src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,281 @@
+# # This file contains the shared helper functions for the polygon clipping functionalities.
+
+#= This is the struct that makes up a_list and b_list. Many values are only used if point is
+an intersection point (ipt). =#
+mutable struct PolyNode{T <: AbstractFloat}
```

Comment:
It is mutable because you don't have all of the information for each node when they are made. Like you don't know where each node is in `b_list` when you are making `a_list`.  Would it be better to make them immutable and then just make a totally new node rather than updating the existing node?

Reply:
Probably worth benchmarking at least